File: Top 100 Codes/01-Getting Started/24-Perfect Square/PerfectSquare.cpp
#include <iostream>
#include <cmath> // For sqrt and ceil/floor functions
using namespace std;

// Function to check if the number is a perfect square
void checkPerfectSquare(int n)
{
    // Calculate the square root of the number
    // If the ceiling and floor of the square root are equal, it means the number is a perfect square
    if (ceil(sqrt(n)) == floor(sqrt(n))) {
        cout << "True"; // The number is a perfect square
    }
    else {
        cout << "False"; // The number is not a perfect square
    }
}

int main()
{
    int n;

    // Take user input for the number
    cout << "Enter a number: ";
    cin >> n;

    // Check if the entered number is a perfect square
    checkPerfectSquare(n);

    return 0;
}

File: Top 100 Codes/01-Getting Started/24-Perfect Square/PerfectSquare.py
import math

# Function to check if a number is a perfect square
def check_perfect_square(x):
    # Check if the square root of x is an integer
    if math.ceil(math.sqrt(x)) == math.floor(math.sqrt(x)):
        print("True")  # The number is a perfect square
    else:
        print("False")  # The number is not a perfect square

# Take user input for the number
n = int(input("Enter a number: "))

# Check if the entered number is a perfect square
check_perfect_square(n)
